controlplane/kubeadm/controllers/scale_test.go

```diff
@@ -208,13 +208,54 @@ func TestKubeadmControlPlaneReconciler_scaleDownControlPlane_NoError(t *testing.
 		g.Expect(fakeClient.List(context.Background(), &controlPlaneMachines)).To(Succeed())
 		g.Expect(controlPlaneMachines.Items).To(HaveLen(0))
 	})
-	t.Run("does not deletes control plane Machine if preflight checks fails", func(t *testing.T) {
+	t.Run("deletes the oldest control plane Machine even if preflight checks fails", func(t *testing.T) {
 		g := NewWithT(t)
 
 		machines := map[string]*clusterv1.Machine{
-			"one": machine("one"),
+			"one":   machine("one", withTimestamp(time.Now().Add(-1*time.Minute))),
+			"two":   machine("two", withTimestamp(time.Now())),
+			"three": machine("three", withTimestamp(time.Now())),
 		}
-		fakeClient := newFakeClient(g, machines["one"])
+		setMachineHealthy(machines["two"])
+		setMachineHealthy(machines["three"])
+		fakeClient := newFakeClient(g, machines["one"], machines["two"], machines["three"])
+
+		r := &KubeadmControlPlaneReconciler{
+			Log:      log.Log,
```

Comment:
Note, I had to add this to avoid an NPE in preflightChecks constructing the logger